sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -356,8 +372,14 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
       }
     }
 
+    val broadcastHint = project match {
+      case p @ Project(projectList, Hint("BROADCAST", tables, child)) =>
+        if (tables.nonEmpty) s"/*+ MAPJOIN(${tables.mkString(", ")}) */" else ""
+      case _ => ""
+    }
     build(
       "SELECT",
+      broadcastHint,
```

Comment:
"SELECT" occurs in the followings. But, I didn't added meaning-logic based on the testcases.
- aggregateToSQL
- generateToSQL  => This has only "SELECT 1"
- groupingSetToSQL
- projectToSQL
- windowToSQL

I think the test coverage enough due to it generates the above cases. If you suggests more testcases, I welcome. I like robustness both for this PR and for the future.